From fed77ffd89ea4501fe7b7103197dc7541246e3bb Mon Sep 17 00:00:00 2001
From: Garrett D'Amore <garrett@damore.org>
Date: Sat, 30 Jul 2022 15:26:57 -0400
Subject: [PATCH] 14871 Retire setuname Reviewed by: Toomas Soome
 <tsoome@me.com> Reviewed by: Peter Tribble <peter.tribble@gmail.com> Reviewed
 by: Andrew Stormont <andyjstormont@gmail.com> Approved by: Joshua M. Clulow
 <josh@sysmgr.org>

---
 usr/src/cmd/Makefile                  |   1 -
 usr/src/cmd/setuname/Makefile         |  59 ---
 usr/src/cmd/setuname/Makefile.com     |  57 ---
 usr/src/cmd/setuname/amd64/Makefile   |  34 --
 usr/src/cmd/setuname/i386/Makefile    |  34 --
 usr/src/cmd/setuname/setuname.c       | 554 --------------------------
 usr/src/cmd/setuname/sparcv9/Makefile |  35 --
 usr/src/lib/libsecdb/exec_attr.txt    |   3 +-
 usr/src/man/man8/Makefile             |   1 -
 usr/src/man/man8/setuname.8           |  74 ----
 usr/src/pkg/manifests/SUNWcs.man8.inc |   1 -
 usr/src/pkg/manifests/SUNWcs.p5m      |   3 -
 12 files changed, 1 insertion(+), 855 deletions(-)
 delete mode 100644 usr/src/cmd/setuname/Makefile
 delete mode 100644 usr/src/cmd/setuname/Makefile.com
 delete mode 100644 usr/src/cmd/setuname/amd64/Makefile
 delete mode 100644 usr/src/cmd/setuname/i386/Makefile
 delete mode 100644 usr/src/cmd/setuname/setuname.c
 delete mode 100644 usr/src/cmd/setuname/sparcv9/Makefile
 delete mode 100644 usr/src/man/man8/setuname.8

diff --git a/usr/src/cmd/Makefile b/usr/src/cmd/Makefile
index c510361f60..e455735102 100644
--- a/usr/src/cmd/Makefile
+++ b/usr/src/cmd/Makefile
@@ -378,7 +378,6 @@ COMMON_SUBDIRS=		\
 	setfacl		\
 	setmnt		\
 	setpgrp		\
-	setuname	\
 	sgs		\
 	sh		\
 	sleep		\
diff --git a/usr/src/cmd/setuname/Makefile b/usr/src/cmd/setuname/Makefile
deleted file mode 100644
index 6f84afa83e..0000000000
--- a/usr/src/cmd/setuname/Makefile
+++ /dev/null
@@ -1,59 +0,0 @@
-#
-# CDDL HEADER START
-#
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
-#
-# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
-#
-#
-#ident	"%Z%%M%	%I%	%E% SMI"
-#
-# Copyright 2003 Sun Microsystems, Inc.  All rights reserved.
-# Use is subject to license terms.
-#
-# cmd/setuname/Makefile
-#
-
-PROG=	setuname
-
-include ../Makefile.cmd
-
-$(64ONLY)SUBDIRS=	$(MACH)
-$(BUILD64)SUBDIRS += $(MACH64)
-
-all	:=	TARGET = all
-install	:=	TARGET = install
-clean	:=	TARGET = clean
-clobber	:=	TARGET = clobber
-lint	:=	TARGET = lint
-
-.KEEP_STATE:
-
-all:	$(SUBDIRS)
-
-clean clobber lint:	$(SUBDIRS)
-
-install:	$(SUBDIRS)
-	-$(RM) $(ROOTPROG)
-	-$(LN) $(ISAEXEC) $(ROOTPROG)
-
-$(SUBDIRS):	FRC
-	@cd $@; pwd; $(MAKE) $(TARGET)
-
-FRC:
-
-include ../Makefile.targ
diff --git a/usr/src/cmd/setuname/Makefile.com b/usr/src/cmd/setuname/Makefile.com
deleted file mode 100644
index f5252a3fb7..0000000000
--- a/usr/src/cmd/setuname/Makefile.com
+++ /dev/null
@@ -1,57 +0,0 @@
-#
-# CDDL HEADER START
-#
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
-#
-# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
-#
-#
-# Copyright 2004 Sun Microsystems, Inc.  All rights reserved.
-# Use is subject to license terms.
-#
-
-PROG= 	setuname
-OBJS=	$(PROG).o
-SRCS=	$(OBJS:%.o=../%.c)
-
-include ../../Makefile.cmd
-
-CFLAGS +=	$(CCVERBOSE)
-CFLAGS64 +=	$(CCVERBOSE)
-CERRWARN +=	-_gcc=-Wno-unused-variable
-CERRWARN +=	-_gcc=-Wno-parentheses
-LDLIBS +=	-lkvm
-
-CLEANFILES	+= $(OBJS)
-
-.KEEP_STATE:
-
-all: $(PROG) 
-
-$(PROG): $(OBJS)
-	$(LINK.c) $(OBJS) -o $@ $(LDLIBS)
-	$(POST_PROCESS)
-
-lint:	lint_SRCS
-
-%.o:	../%.c
-	$(COMPILE.c) $<
-
-clean:
-	$(RM) $(CLEANFILES)
-
-include ../../Makefile.targ
diff --git a/usr/src/cmd/setuname/amd64/Makefile b/usr/src/cmd/setuname/amd64/Makefile
deleted file mode 100644
index 29e8ab7aae..0000000000
--- a/usr/src/cmd/setuname/amd64/Makefile
+++ /dev/null
@@ -1,34 +0,0 @@
-#
-# CDDL HEADER START
-#
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
-#
-# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
-#
-#
-# Copyright 2004 Sun Microsystems, Inc.  All rights reserved.
-# Use is subject to license terms.
-#
-# ident	"%Z%%M%	%I%	%E% SMI"
-#
-
-.KEEP_STATE:
-
-include ../Makefile.com
-include ../../Makefile.cmd.64
-
-install: all $(ROOTPROG64)
diff --git a/usr/src/cmd/setuname/i386/Makefile b/usr/src/cmd/setuname/i386/Makefile
deleted file mode 100644
index 8b456406ad..0000000000
--- a/usr/src/cmd/setuname/i386/Makefile
+++ /dev/null
@@ -1,34 +0,0 @@
-#
-# CDDL HEADER START
-#
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
-#
-# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
-#
-#
-#ident	"%Z%%M%	%I%	%E% SMI"
-#
-# Copyright (c) 1997, by Sun Microsystems, Inc.
-# All rights reserved.
-#
-# cmd/setuname/i386/Makefile
-
-.KEEP_STATE:
-
-include ../Makefile.com
-
-install: all $(ROOTPROG32)
diff --git a/usr/src/cmd/setuname/setuname.c b/usr/src/cmd/setuname/setuname.c
deleted file mode 100644
index 69434e6d2d..0000000000
--- a/usr/src/cmd/setuname/setuname.c
+++ /dev/null
@@ -1,554 +0,0 @@
-/*
- * CDDL HEADER START
- *
- * The contents of this file are subject to the terms of the
- * Common Development and Distribution License (the "License").
- * You may not use this file except in compliance with the License.
- *
- * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
- * or http://www.opensolaris.org/os/licensing.
- * See the License for the specific language governing permissions
- * and limitations under the License.
- *
- * When distributing Covered Code, include this CDDL HEADER in each
- * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
- * If applicable, add the following below this CDDL HEADER, with the
- * fields enclosed by brackets "[]" replaced with your own identifying
- * information: Portions Copyright [yyyy] [name of copyright owner]
- *
- * CDDL HEADER END
- */
-/*	Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T	*/
-/*	  All Rights Reserved  	*/
-
-
-/*
- * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
- * Use is subject to license terms.
- */
-
-/*
- * Copyright (c) 2018, Joyent, Inc.
- */
-
-/*
- *  setuname [-t] [-s name] [-n node] 
- */
-
-/*
- *  Header files referenced:
- *	<stdio.h>	Standard I/O 
- *	<unistd.h>	Standard UNIX definitions
- *	<string.h>	String handling 
- *	<fmtmsg.h>	Standard message generation 
- *	<ctype.h>	Character types
- *	<errno.h>	Error handling
- *	<signal.h>	Signal handling 
- *	<sys/types.h>	Data types
- *	<sys/fcntl.h>	File control
- *	<sys/utsname.h>	System Name
- *	<sys/sys3b.h>	sys3b() definitions
- *	<nlist.h>	Definitions for Sun symbol table entries
- */
-
-#include	<stdio.h>
-#include	<unistd.h>
-#include	<string.h>
-#include	<fmtmsg.h>
-#include	<ctype.h>
-#include	<errno.h>
-#include	<signal.h>
-#include	<sys/types.h>
-#include	<sys/uio.h>
-#include	<sys/fcntl.h>
-#include	<sys/psw.h>
-#include	<sys/utsname.h>
-
-#if u3b || u3b15 || u3b2
-#include	<sys/sys3b.h>
-#endif
-
-#if sun
-#include	<nlist.h>
-#include	<kvm.h>
-#endif
-
-/*
- * Externals referenced (and not defined in a header)
- *	optind		index to the next arg for getopt()
- *	opterr		FLAG, TRUE tells getopt() to write messages
- *	optarg		Ptr to an option's argument
- *	getopt()	Gets an option from the command line
- *	putenv()	Writes values into the environment
- *	exit()		Exit the process
- *	access()	Check accessibility of a file
- *	malloc()	Allocate a block of main memory
- *	free()		Free allocated space
- *	lseek()		Seek within a file 
- *	open()		Open a file
- *	close()		Close an open file
- */
- 
-extern	int		optind;		/* argv[] index of next arg */
-extern	int		opterr;		/* TRUE if getopt() is to print msgs */
-extern	char	       *optarg;		/* Argument to parsed option */
-extern	int		getopt();	/* Get an option from the command line */
-extern	int	       	putenv();	/* Put a value into the environment */
-extern	void		exit();		/* Exit the process */
-extern	int		access();	/* Check the accessibility of a file */
-extern	void	       *malloc();	/* Get a chunk of main memory */
-extern	void		free();		/* Free alloc'd space */
-extern	long		lseek();	/* Seek within a file */
-extern	int		open();		/* Open a file */
-extern	int		close();	/* Close an open a file */
-
-/*
- *  L O C A L   D E F I N I T I O N S
- */
-
-/*
- * Constants 
- */
-
-#ifndef	TRUE
-#define	TRUE		(1)
-#endif
-
-#ifndef	FALSE
-#define	FALSE		(0)
-#endif
-
-#ifndef	NULL
-#define	NULL		(0)
-#endif
-
-#define	OPTSTRING	"tn:s:"
-
-#define	EX_OK		0
-#define	EX_ERROR	1
-
-#define	RC_FILENAME	"/etc/rc2.d/S18setuname"
-#define RC_DIRNAME	"/etc/rc2.d"
-
-
-/*
- *  Messages
- */
-
-#define	E_USAGE		"usage: setuname [-t] [-s name] [-n node]"
-#define	E_MISSING	"Either -s name or -n node must be specified"
-#define	E_UNAME		"Unable to get existing uname values"
-#define E_INVNAME	"System-name invalid: %s"
-#define E_LONGNAME	"System-name too long: %s"
-#define E_INVNODE	"Network node-name invalid: %s"
-#define E_LONGNODE	"Network node-name too long: %s"
-#define E_NOPERMS	"No permissions, request denied"
-#define E_NOSUCHDIR	"Directory doesn't exist: %s"
-#define	E_INTERNAL	"Internal error: %d"
-
-/*
- * Macros:
- *	stdmsg(r,l,s,t)	    Write a standard message.  
- *				'r' is the recoverability flag
- *				'l' is the label
- *				's' is the severity 
- *				't' is the text.
- *	strend(p)	    Return the address of the end of a string
- *			    (This is supposed to be defined in <sys/inline.h>
- *			    but that file has string-handing def'ns that
- *			    conflict with <string.h>, so we can't use it!
- *			    MR dn89-04701 requests this fix.
- */
- 
-#define	stdmsg(r,l,s,t)	(void) fmtmsg(MM_PRINT|MM_UTIL|r,l,s,t,MM_NULLACT,MM_NULLTAG)
-#define strend(p)       strrchr(p,'\0')
-
-/*
- * Local functions:
- *	setuname	Changes the system name and the network node name 
- */
-
-static int	setuname();		/* This does the "real" work */
-
-
-/*
- * Local data
- *	lbl		Buffer for the standard message label
- *	txt		Buffer for the standard message text
- */
- 
-static	char		lbl[MM_MXLABELLN+1];	/* Space for std msg label */
-static	char		msg[MM_MXTXTLN+1];	/* Space for std msg text  */
-
-/*
- *  int main(argc, argv)
- *	int	argc
- *	char   *argv;
- */
-
-int 
-main(argc, argv)
-	int	argc;			/* Argument count  */
-	char   *argv[];			/* Argument vector */
-{
-	/* Automatic data */
-	char	       *n_arg;			/* Ptr to arg for -n */
-	char	       *s_arg;			/* Ptr to arg for -s */
-	int		t_seen;			/* FLAG, -t option seen */
-	char	       *cmdname;		/* Ptr to the command's name */
-	char	       *p;			/* Temp pointer */
-	int		usageerr;		/* FLAG, TRUE if usage error */
-	int		exitcode;		/* Value to exit with */
-	int		c;			/* Temp character */
-	int		ok;			/* Flag, everything okay? */
-
-	/* Build the standard-message label */
-	if (p = strrchr(argv[0], '/')) cmdname = p+1;
-	else cmdname = argv[0];
-	(void) strcat(strcpy(lbl, "UX:"), cmdname);
-
-	/* Make only the text in standard messages appear (SVR4.0 only) */
-	(void) putenv("MSGVERB=text");
-
-	
-	/* Initializations */
-	n_arg = s_arg = (char *) NULL;
-	t_seen = FALSE;
-
-
-	/* 
-	 * Parse command 
-	 */
-
-	usageerr = FALSE;
-	opterr = FALSE;
-	while (!usageerr && (c = getopt(argc, argv, OPTSTRING)) != EOF) switch(c) {
-
-	case 'n':			/* -n node */
-	    if (n_arg) usageerr = TRUE;
-	    else n_arg = optarg;
-	    break;
-
-	case 's':			/* -s name */
-	    if (s_arg) usageerr = TRUE;
-	    else s_arg = optarg;
-	    break;
-
-	case 't':			/* -t */
-	    if (t_seen) usageerr = TRUE;
-	    else t_seen = TRUE;
-	    break;
-	    
-	default:			/* Something that doesn't exist */
-	    usageerr = TRUE;
-	}   /* switch() */
-
-	/* If there was a usage error, report the error and exit */
-	if ((argc >= (optind+1)) || usageerr) {
-	    stdmsg(MM_NRECOV, lbl, MM_ERROR, E_USAGE);
-	    exit(EX_ERROR);
-	}
-
-	/* Either -n <node> or -s <name> has to be specified */
-	if (!(n_arg || s_arg)) {
-	    stdmsg(MM_NRECOV, lbl, MM_ERROR, E_MISSING);
-	    exit(EX_ERROR);
-	}
-
-
-	/* 
-	 * Validate arguments:
-	 *  - The length of the system name must be less than SYS_NMLN-1
-	 *    characters,
-	 *  - The length of the network node-name must be less than 
-	 *    SYS_NMLN-1 characters,
-	 *  - The system name must equal [a-zA-Z0-9-_]+,
-	 *  - The network node-name must equal [a-zA-Z0-9-_]+.
-	 */
-
-	/* Check the length and the character-set of the system name */
-	if (s_arg) {
-
-	    /* Check length of the system name */
-	    if (strlen(s_arg) > (size_t)(SYS_NMLN-1)) {
-		(void) sprintf(msg, E_LONGNAME, s_arg);
-		stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-		exit(EX_ERROR);
-	    }
-
-	    /* Check the character-set */
-	    ok = TRUE;
-	    for (p = s_arg ; ok && *p ; p++) {
-		if (!isalnum(*p) && (*p != '-') && (*p != '_')) ok = FALSE;
-	    }
-	    if (!ok || (p == s_arg)) {
-		(void) sprintf(msg, E_INVNAME, s_arg);
-		stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-		exit(EX_ERROR);
-	    }
-	}
-
-	/* Check the length and the character-set of the network node-name */
-
-	if (n_arg) {
-
-	    /* Check length of the network node-name */
-	    if (strlen(n_arg) > (size_t)(SYS_NMLN-1)) {
-		(void) sprintf(msg, E_LONGNODE, n_arg);
-		stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-		exit(EX_ERROR);
-	    }
-
-	    /* Check the character-set */
-	    ok = TRUE;
-	    for (p = n_arg ; ok && *p ; p++) {
-		if (!isalnum(*p) && (*p != '-') && (*p != '_')) ok = FALSE;
-	    }
-	    if (!ok || (p == n_arg)) {
-		(void) sprintf(msg, E_INVNODE, n_arg);
-		stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-		exit(EX_ERROR);
-	    }
-	}
-
-
-	/*
-	 * Make sure we have access to needed resources:
-	 *   -  Read/write access to kernel memory (/dev/kmem)
-	 *   -  If -t is not specified, read/write access to /etc/rc2.d
-	 *   -  If -t is not specified, read access to /etc/rc2.d/S18setuname
-	 */
-	
-	if (access("/dev/kmem", R_OK|W_OK) == 0) {
-	    if (access(RC_DIRNAME, R_OK|W_OK) == 0) {
-		if ((access(RC_FILENAME, R_OK) != 0) && 
-		    (access(RC_FILENAME, F_OK) == 0)) {
-		    stdmsg(MM_NRECOV, lbl, MM_ERROR, E_NOPERMS);
-		    exit(EX_ERROR);
-		}
-	    } 
-	    else {
-		if (access(RC_DIRNAME, F_OK) == 0) {
-		    stdmsg(MM_NRECOV, lbl, MM_ERROR, E_NOPERMS);
-		    exit(EX_ERROR);
-		} 
-		else {
-		    (void) sprintf(msg, E_NOSUCHDIR, RC_DIRNAME);
-		    stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-		    exit(EX_ERROR);
-		}
-	    }
-	} 
-	else {
-	    stdmsg(MM_NRECOV, lbl, MM_ERROR, E_NOPERMS);
-	    exit(EX_ERROR);
-	}
-
-
-	/* Attempt the setuname */
-	if (setuname(t_seen, s_arg, n_arg) == 0) exitcode = EX_OK;
-	else {
-	    (void) sprintf(msg, E_INTERNAL, errno);
-	    stdmsg(MM_NRECOV, lbl, MM_ERROR, msg);
-	    exitcode = EX_ERROR;
-	}
-
-	/* Finished */
-	return (exitcode);
-}  /* main() */
-
-/*
- * int setuname(temp, name, node)
- *	int	temp
- *	char   *name
- *	char   *node
- *
- *	Set any or all of the following machine parameters, either
- *	temporarily or permanently, depending on <temp>.
- *	    - System name
- *	    - Network Node-name
- */
-
-static int 
-setuname(temp, sysname, nodename)
-	int	temp;		/* Set in kernel only flag */
-	char   *sysname;	/* System name */
-	char   *nodename;	/* Network node-name */
-{
-	/* Automatic Data */
-	struct utsname	utsname;	/* Space for the kernel's utsname information */
-#if u3b || u3b15 || u3b2
-	struct s3bsym  *symbtbl;	/* The kernel's symbol table */
-#endif
-#if sun
-	struct nlist nl[] = {
-		{"utsname", 0, 0, 0, 0, 0},
-		{NULL}
-	};
-	kvm_t *kd;
-#endif
-	uintptr_t	utsname_addr;	/* Addr of "utsname" in the kernel */
-	char	       *sysnm = (char *)NULL;		/* System name to set (from file or arg) */
-	char	       *nodenm = (char *)NULL;		/* Network node-name to set (from file or arg) */
-	FILE	       *fd;		/* Std I/O File Descriptor for /etc/rc2.d/S18setuname */
-	char	       *p;		/* Temp pointer */
-	void	      (*oldsighup)();	/* Function to call for SIGHUP */
-	void	      (*oldsigint)();	/* Function to call for SIGINT */
-	int		rtncd;		/* Value to return to the caller */
-	unsigned long	symbtblsz;	/* The size of the kernel's symbol table, in bytes */
-	int		memfd;		/* File descriptor:  open kernel memory */
-	int		i;		/* Temp counter */	
-
-
-	/* Nothing's gone wrong yet (but we've only just begun!) */
-	rtncd = 0;
-
-
-	/*
-	 * Get the virtual address of the symbol "utsname" in the kernel
-	 * so we can get set the system name and/or the network node-name
-	 * directly in the kernel's memory space.
-	 */
-
-#if u3b || u3b15 || u3b2
-	if ((sys3b(S3BSYM, (struct s3bsym *) &symbtblsz, sizeof(symbtblsz)) == 0) &&
-	    (symbtbl = (struct s3bsym *) malloc(symbtblsz))) {
-
-	    (void) sys3b(S3BSYM, symbtbl, symbtblsz);
-	    p = (char *) symbtbl;
-	    for (i = symbtbl->count; i-- && (strcmp(p, "utsname") != 0) ; p = S3BNXTSYM(p)) ;
-	    if (i >= 0) utsname_addr = S3BSVAL(p);
-	    else rtncd = -1;
-	    free((void *) symbtbl);
-
-	} else rtncd = -1;
-
-#elif sun
-        /* Check out namelist and memory files. */
-	if ((kd = kvm_open(NULL, NULL, NULL, O_RDWR, NULL)) == NULL)
-		rtncd = -1;
-	else if (kvm_nlist(kd, nl) != 0)
-		rtncd = -1;
-	else if (nl[0].n_value == 0)
-		rtncd = -1;
-	else
-		utsname_addr = (uintptr_t)nl[0].n_value;
-#else
-	if (nlist("/unix", nl) != 0)
-		rtncd = -1;
-#endif
-	if (rtncd != 0) return(rtncd);
-
-	/* 
-	 * Open the kernel's memory, get the existing "utsname" structure,
-	 * change the system name and/or the network node-name in that struct,
-	 * write it back out to kernel memory, then close kernel memory.
-	 */
-#ifdef sun
-	if (kvm_kread(kd, utsname_addr, &utsname, sizeof (utsname)) ==
-	    sizeof (utsname)) {
-		if (sysname)
-			(void) strncpy(utsname.sysname, sysname,
-			    sizeof (utsname.sysname));
-		if (nodename)
-			(void) strncpy(utsname.nodename, nodename,
-			    sizeof (utsname.nodename));
-		(void) kvm_kwrite(kd, utsname_addr, &utsname, sizeof (utsname));
-		(void) kvm_close(kd);
-	} else
-		return (-1);
-#else /* sun */
-	if ((memfd = open("/dev/kmem", O_RDWR, 0)) > 0) {
-	    if ((lseek(memfd, (long) utsname_addr, SEEK_SET) != -1) && 
-		(read(memfd, &utsname, sizeof(utsname)) == sizeof(utsname))) {
-		if (sysname) (void) strncpy(utsname.sysname, sysname, sizeof(utsname.sysname));
-		if (nodename) (void) strncpy(utsname.nodename, nodename, sizeof(utsname.nodename));
-		(void) lseek(memfd, (long) utsname_addr, SEEK_SET);
-		(void) write(memfd, &utsname, sizeof(utsname));
-		(void) close(memfd);
-	    } else rtncd = -1;
-	} else rtncd = -1;
-	if (rtncd != 0) return(rtncd);
-#endif /* sun */
-
-
-	/*
-	 * If the "temp" flag is FALSE, we need to permanently set the
-	 * system name in the file  /etc/rc2.d/S18setuname
-	 */
-
-	if (!temp) {
-	    /* 
-	     * If a name was specified by the caller, use that, otherwise, use
-	     * whatever was in the "rc" file.
-	     */
-
-		if (sysname) sysnm = sysname;
-		if (nodename) nodenm = nodename;
-
-
-	    /* 
-	     * Write the file /etc/rc2.d/S18setuname so that the system name is
-	     * set on boots and state changes.  
-	     *
-	     * DISABLED SIGNALS: SIGHUP, SIGINT
-	     */
-
-	    /* Give us a reasonable chance to complete without interruptions */
-		oldsighup = signal(SIGHUP, SIG_IGN);
-		oldsigint = signal(SIGINT, SIG_IGN);
-
-	    /* Write the new setuname "rc" file */
-		if (sysname != NULL) {
-			if ((fd = fopen(RC_FILENAME, "w")) != (FILE *) NULL) {
-				(void) fprintf(fd, "# %s\n", sysnm);
-				(void) fprintf(fd, "#\n");
-				(void) fprintf(fd, "# This script, generated by the setuname command,\n");
-				(void) fprintf(fd, "# sets the system's system-name\n");
-				(void) fprintf(fd, "#\n");
-				if (sysnm && *sysnm)
-					(void) fprintf(fd, "setuname -t -s %s\n", sysnm);
-				(void) fclose(fd);
-			} else return(rtncd = -1);
-		}
-
-		if(nodename != NULL) {
-			char curname[SYS_NMLN];
-			int curlen;
-			FILE *file;
-
-			if ((file = fopen("/etc/nodename", "r")) != NULL) {
-				curlen = fread(curname, sizeof(char), SYS_NMLN, file);
-				for (i = 0; i < curlen; i++) {
-					if (curname[i] == '\n') {
-						curname[i] = '\0';
-						break;
-					}
-				}
-				if (i == curlen) {
-					curname[curlen] = '\0';
-				}
-				(void)fclose(file);
-			} else {
-				curname[0] = '\0';
-			}
-			if (strcmp(curname, nodenm) != 0) {
-				if ((file = fopen("/etc/nodename", "w")) == NULL) {
-					(void) fprintf(stderr, "setuname: error in writing name\n");
-					exit(1);
-				} 
-				if (fprintf(file, "%s\n", nodenm) < 0) {
-					(void) fprintf(stderr, "setuname: error in writing name\n");
-					exit(1);
-				}
-				(void)fclose(file);
-			}		
-		}
-	    /* Restore signal handling */
-		(void) signal(SIGHUP, oldsighup);
-		(void) signal(SIGINT, oldsigint);
-	}	/* if (!temp) */
-
-	/* Fini */
-	return(rtncd);
-}
diff --git a/usr/src/cmd/setuname/sparcv9/Makefile b/usr/src/cmd/setuname/sparcv9/Makefile
deleted file mode 100644
index 87afd4d6b0..0000000000
--- a/usr/src/cmd/setuname/sparcv9/Makefile
+++ /dev/null
@@ -1,35 +0,0 @@
-#
-# CDDL HEADER START
-#
-# The contents of this file are subject to the terms of the
-# Common Development and Distribution License, Version 1.0 only
-# (the "License").  You may not use this file except in compliance
-# with the License.
-#
-# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
-# or http://www.opensolaris.org/os/licensing.
-# See the License for the specific language governing permissions
-# and limitations under the License.
-#
-# When distributing Covered Code, include this CDDL HEADER in each
-# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
-# If applicable, add the following below this CDDL HEADER, with the
-# fields enclosed by brackets "[]" replaced with your own identifying
-# information: Portions Copyright [yyyy] [name of copyright owner]
-#
-# CDDL HEADER END
-#
-#
-#ident	"%Z%%M%	%I%	%E% SMI"
-#
-# Copyright (c) 1997, by Sun Microsystems, Inc.
-# All rights reserved.
-#
-# cmd/setuname/sparcv9/Makefile
-
-.KEEP_STATE:
-
-include ../Makefile.com
-include ../../Makefile.cmd.64
-
-install: all $(ROOTPROG64)
diff --git a/usr/src/lib/libsecdb/exec_attr.txt b/usr/src/lib/libsecdb/exec_attr.txt
index feb95dd8c4..14d7dc9ac3 100644
--- a/usr/src/lib/libsecdb/exec_attr.txt
+++ b/usr/src/lib/libsecdb/exec_attr.txt
@@ -193,7 +193,6 @@ Network Management:solaris:cmd:::/sbin/ipadm:euid=netadm;egid=netadm;\
 Network Management:suser:cmd:::/usr/bin/netstat:uid=0
 Network Management:suser:cmd:::/usr/bin/rup:euid=0
 Network Management:suser:cmd:::/usr/bin/ruptime:euid=0
-Network Management:suser:cmd:::/usr/bin/setuname:euid=0
 Network Management:suser:cmd:::/usr/sbin/asppp2pppd:euid=0
 Network Management:suser:cmd:::/usr/sbin/ifconfig:uid=0
 Network Management:suser:cmd:::/usr/sbin/ipaddrsel:euid=0
@@ -236,7 +235,7 @@ Object Access Management:suser:cmd:::/usr/bin/chown:euid=0
 Object Access Management:suser:cmd:::/usr/bin/getfacl:euid=0
 Object Access Management:suser:cmd:::/usr/bin/setfacl:euid=0
 Primary Administrator:solaris:cmd:::*:uid=0;gid=0
-Printer Management:suser:cmd:::/usr/lib/lp/local/lpadmin:uid=lp;gid=lp 
+Printer Management:suser:cmd:::/usr/lib/lp/local/lpadmin:uid=lp;gid=lp
 Printer Management:suser:cmd:::/usr/sbin/lpfilter:euid=lp;uid=lp
 Printer Management:suser:cmd:::/usr/sbin/lpforms:euid=lp
 Printer Management:suser:cmd:::/usr/sbin/lpusers:euid=lp
diff --git a/usr/src/man/man8/Makefile b/usr/src/man/man8/Makefile
index e978668679..8d683fd99f 100644
--- a/usr/src/man/man8/Makefile
+++ b/usr/src/man/man8/Makefile
@@ -429,7 +429,6 @@ _MANFILES=	6to4relay.8		\
 		scadm.8			\
 		sdpadm.8		\
 		sendmail.8		\
-		setuname.8		\
 		share.8			\
 		share_nfs.8		\
 		shareall.8		\
diff --git a/usr/src/man/man8/setuname.8 b/usr/src/man/man8/setuname.8
deleted file mode 100644
index 00d6fc1471..0000000000
--- a/usr/src/man/man8/setuname.8
+++ /dev/null
@@ -1,74 +0,0 @@
-'\" te
-.\"  Copyright 1989 AT&T Copyright (c) 1998 Sun Microsystems, Inc. All Rights Reserved.
-.\" The contents of this file are subject to the terms of the Common Development and Distribution License (the "License").  You may not use this file except in compliance with the License.
-.\" You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE or http://www.opensolaris.org/os/licensing.  See the License for the specific language governing permissions and limitations under the License.
-.\" When distributing Covered Code, include this CDDL HEADER in each file and include the License file at usr/src/OPENSOLARIS.LICENSE.  If applicable, add the following below this CDDL HEADER, with the fields enclosed by brackets "[]" replaced with your own identifying information: Portions Copyright [yyyy] [name of copyright owner]
-.TH SETUNAME 8 "Jul 9, 2003"
-.SH NAME
-setuname \- change machine information
-.SH SYNOPSIS
-.LP
-.nf
-\fBsetuname\fR [\fB-t\fR] [\fB-n\fR \fInode\fR] [\fB-s\fR \fIname\fR]
-.fi
-
-.SH DESCRIPTION
-.sp
-.LP
-The \fBsetuname\fR utility changes the parameter value for the system name and
-node name. Each parameter can be changed using \fBsetuname\fR and the
-appropriate option.
-.sp
-.LP
-Either or both the \fB-s\fR and \fB-n\fR options must be given when invoking
-\fBsetuname\fR.
-.sp
-.LP
-The system architecture may place requirements on the size of the system and
-network node name. The command will issue a fatal warning message and an error
-message if the name entered is incompatible with the system requirements.
-.SH OPTIONS
-.sp
-.LP
-The following options are supported:
-.sp
-.ne 2
-.na
-\fB\fB-n\fR\fI node\fR\fR
-.ad
-.RS 11n
-Changes the node name. \fInode\fR specifies the new network node name and can
-consist of alphanumeric characters and the special characters dash, underbar,
-and dollar sign.
-.RE
-
-.sp
-.ne 2
-.na
-\fB\fB-s\fR\fI name\fR\fR
-.ad
-.RS 11n
-Changes the system name. \fIname\fR specifies new system name and can consist
-of alphanumeric characters and the special characters dash, underbar, and
-dollar sign.
-.RE
-
-.sp
-.ne 2
-.na
-\fB\fB-t\fR\fR
-.ad
-.RS 11n
-Temporary change. No attempt will be made to create a permanent change.
-.RE
-
-.SH SEE ALSO
-.sp
-.LP
-.BR attributes (7)
-.SH NOTES
-.sp
-.LP
-\fBsetuname\fR attempts to change the parameter values in two places: the
-running kernel and, as necessary per implementation, to cross system reboots. A
-temporary change changes only the running kernel.
diff --git a/usr/src/pkg/manifests/SUNWcs.man8.inc b/usr/src/pkg/manifests/SUNWcs.man8.inc
index e01e313a47..89c67f870f 100644
--- a/usr/src/pkg/manifests/SUNWcs.man8.inc
+++ b/usr/src/pkg/manifests/SUNWcs.man8.inc
@@ -228,7 +228,6 @@ file path=usr/share/man/man8/sac.8
 file path=usr/share/man/man8/sacadm.8
 file path=usr/share/man/man8/saf.8
 file path=usr/share/man/man8/savecore.8
-file path=usr/share/man/man8/setuname.8
 file path=usr/share/man/man8/share.8
 file path=usr/share/man/man8/share_nfs.8
 file path=usr/share/man/man8/shareall.8
diff --git a/usr/src/pkg/manifests/SUNWcs.p5m b/usr/src/pkg/manifests/SUNWcs.p5m
index b9b55d0d89..918e50c3a0 100644
--- a/usr/src/pkg/manifests/SUNWcs.p5m
+++ b/usr/src/pkg/manifests/SUNWcs.p5m
@@ -540,7 +540,6 @@ $(i386_ONLY)file path=usr/bin/$(ARCH32)/nohup mode=0555
 $(i386_ONLY)file path=usr/bin/$(ARCH32)/prctl mode=0555
 $(i386_ONLY)file path=usr/bin/$(ARCH32)/prstat mode=0555
 file path=usr/bin/$(ARCH32)/savecore mode=0555
-$(i386_ONLY)file path=usr/bin/$(ARCH32)/setuname mode=0555
 dir  path=usr/bin/$(ARCH64)
 file path=usr/bin/$(ARCH64)/amt mode=0555
 file path=usr/bin/$(ARCH64)/crle mode=0555
@@ -555,7 +554,6 @@ file path=usr/bin/$(ARCH64)/nohup mode=0555
 file path=usr/bin/$(ARCH64)/prctl mode=0555
 file path=usr/bin/$(ARCH64)/prstat mode=0555
 file path=usr/bin/$(ARCH64)/savecore mode=0555
-file path=usr/bin/$(ARCH64)/setuname mode=0555
 $(i386_ONLY)file path=usr/bin/addbadsec mode=0555
 file path=usr/bin/amt mode=0555
 file path=usr/bin/arch mode=0555
@@ -723,7 +721,6 @@ file path=usr/bin/sed mode=0555
 file path=usr/bin/setfacl mode=0555
 file path=usr/bin/setpgrp group=sys mode=0555
 file path=usr/bin/settime mode=0555
-hardlink path=usr/bin/setuname target=../../usr/lib/isaexec
 file path=usr/bin/sleep mode=0555
 file path=usr/bin/strchg group=root mode=0555
 link path=usr/bin/strclean target=../sbin/strclean
-- 
2.42.0

